File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curSum = 0
        maxSum = float('-inf')
        for i in range(len(nums)):
            temp = curSum + nums[i]
            if temp<nums[i]:
                curSum = nums[i]
            else:
                curSum = temp
            if maxSum<curSum:
                maxSum = curSum
        return maxSum
        

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curSum = 0;
        int maxSum = Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            int temp = curSum + nums[i];
            if(temp<nums[i]){
                curSum = nums[i];
            }
            else{
                curSum = temp;
            }
            if(curSum > maxSum){
                maxSum = curSum;
            }
        }
        return maxSum;
    }
}
